src/core/pages/PatientSummary/DialogWithStyles.js

```diff
@@ -0,0 +1,158 @@
+import React, { Component } from "react";
+import { get } from "lodash";
+import compose from 'recompose/compose';
+import { connect } from 'react-redux';
+
+import { withStyles } from "@material-ui/core/styles";
+import Dialog from '@material-ui/core/Dialog';
+import Divider from '@material-ui/core/Divider';
+import Checkbox from '@material-ui/core/Checkbox';
+import Radio from '@material-ui/core/Radio';
+import Typography from "@material-ui/core/Typography";
+
+import { showModeAction } from "../../actions/showModeAction";
+import { showHeadingsAction } from "../../actions/showHeadingsAction";
+import { synopsisData, showModesArray, SHOW_ALL, getHeadingsLists } from "./config";
+
+const styles = {
+    dialogBlock: {
+        minHeight: "300px",
+        minWidth: "500px",
+        margin: "20px",
+    },
+    dialogItem: {
+        height: "100px",
+    },
+    dialogItemColumn: {
+        display: "flex",
+        flexDirection: "column",
+    },
+    dialogLabel: {
+        display: "inline-block",
+        minWidth: "200px",
+    }
+};
+
+/**
+ * This component returns content and functionality of dialog window
+ *
+ * @author Bogdan Shcherban <bsc@piogroup.net>
+ */
+class DialogContent extends Component {
+
+    state = {
+        selectedMode: this.props.showMode ? this.props.showMode : SHOW_ALL,
+        selectedHeadings: this.props.showHeadings ? Object.values(this.props.showHeadings) : getHeadingsLists(),
+    };
+
+    /**
+     * This function checks is current heading was checked
+     *
+     * @author Bogdan Shcherban <bsc@piogroup.net>
+     * @param {string} itemName
+     * @return {boolean}
+     */
+    isHeadingChecked = itemName => {
+        const { selectedHeadings } = this.state;
+        return Object.values(selectedHeadings).indexOf(itemName) !== -1;
+    };
+
+    /**
+     * This function saves headings to select
+     *
+     * @author Bogdan Shcherban <bsc@piogroup.net>
+     * @param {string} itemName
+     */
+    selectHeading = itemName => {
+        const { selectedHeadings } = this.state;
+        let headingsArray = selectedHeadings;
+        if (this.isHeadingChecked(itemName)) {
+            let index = headingsArray.indexOf(itemName);
+            headingsArray.splice(index, 1)
+        } else {
+            headingsArray.push(itemName);
+        }
+        this.setState({
```

Comment:
It's a bad practice to use `setState` based on inspection of the `state` the way you did, because react can roll on multiple state updates, and you may end up in a race condition.

See https://reactjs.org/docs/state-and-lifecycle.html#state-updates-may-be-asynchronous

Prefer calling `setState` as follows:

```
this.setState(state => {
        const { selectedHeadings } = state;
        let headingsArray = selectedHeadings;
        if (Object.values(selectedHeadings).indexOf(itemName) !== -1) {
            let index = headingsArray.indexOf(itemName);
            headingsArray.splice(index, 1)
        } else {
            headingsArray.push(itemName);
        }
        return { selectedHeadings: headingsArray }
})
```

Reply:
Done